server/athenian/api/internal/logical_repos.py

```diff
@@ -43,3 +47,52 @@ def extract_logical_repo(repo: str, offset: int = 2) -> str:
         return repo.split("/", offset)[offset]
     except IndexError:
         return ""
+
+
+@dataclasses.dataclass(frozen=True)
+class RepoName:
+    """The name of a repository.
+
+    Possible formats:
+
+    - repository name: "owner/reponame"
+    - logical repository name: "owner/reponame/logicalname".
+    - prefixed repository name: "provider.com/owner/reponame"
+    - prefixed logical repository name: "provider.com/owner/reponame/logicalname"
+
```

Comment:
We denote "name" as "physical" and "logical_suffix" as "logical". The "prefix" example is "github.com".